scripts/generate-bindings.sh

```diff
@@ -10,29 +10,24 @@ if [[ -z "${SOLC_BIN}" ]]; then
 	SOLC_BIN=solc
 fi
 
-"${SOLC_BIN}" --abi ethereum/contracts/SwapFactory.sol -o ethereum/abi/ --overwrite
-"${SOLC_BIN}" --bin ethereum/contracts/SwapFactory.sol -o ethereum/bin/ --overwrite
-
-"${ABIGEN}" \
-	--abi ethereum/abi/SwapFactory.abi \
-	--bin ethereum/bin/SwapFactory.bin \
-	--pkg contracts \
-	--type SwapFactory \
-	--out ethereum/swap_factory.go
-
-"${SOLC_BIN}" --abi ethereum/contracts/ERC20Mock.sol -o ethereum/abi/ --overwrite
-"${SOLC_BIN}" --bin ethereum/contracts/ERC20Mock.sol -o ethereum/bin/ --overwrite
-
-"${ABIGEN}" \
-	--abi ethereum/abi/ERC20Mock.abi \
-	--bin ethereum/bin/ERC20Mock.bin \
-	--pkg contracts \
-	--type ERC20Mock \
-	--out ethereum/erc20_mock.go
-
-"${SOLC_BIN}" --abi ethereum/contracts/IERC20Metadata.sol -o ethereum/abi/ --overwrite
-"${ABIGEN}" \
-	--abi ethereum/abi/IERC20Metadata.abi \
-	--pkg contracts \
-	--type IERC20 \
-	--out ethereum/ierc20.go
+compile-contract() {
+	local solidity_type_name="${1:?}"
+	local go_type_name="${2:?}"
+	local go_file_name="${3:?}"
+
+	echo "Generating go bindings for ${solidity_type_name}"
+
+	"${SOLC_BIN}" --abi "ethereum/contracts/${solidity_type_name}.sol" -o ethereum/abi/ --overwrite
+	"${SOLC_BIN}" --bin "ethereum/contracts/${solidity_type_name}.sol" -o ethereum/bin/ --overwrite
+	"${ABIGEN}" \
+		--abi "ethereum/abi/${solidity_type_name}.abi" \
+		--bin "ethereum/bin/${solidity_type_name}.bin" \
+		--pkg contracts \
+		--type "${go_type_name}" \
+		--out "ethereum/${go_file_name}.go"
```

Comment:
I didn't want to do it in this PR, but I could easily update this to use the official solc docker images to compile the contract and then developers would not need a specific version of `solc` installed if that sounds like a good idea.

Reply:
does the solc docker image automatically handle versioning? ie will it use the right version for a given contract?

Reply:
They use separate docker containers for the different solc versions.  All the containers are in the same dockerhub repo, but you set the docker image tag to the solc version you want to use. If you need to dynamicly pick between different solc versions, you can parse the needed version from the pragma in the solidity file, and then use that version as the image tag when invoking docker. Doing that only makes sense if you need to compile different contracts with different versions though.

Here's some sample code from a personal project that I created:
```
	solidity_version="$(
		grep --max-count=1 '^pragma solidity' "${solidity_file_name}" |
			sed 's/.*\([[:digit:]]\+\.[[:digit:]]\+\.[[:digit:]]\+\).*/\1/'
	)"
	local solc_exec=(
		docker run --rm
		--volume "${PWD}/contracts:/contracts"
		-u "$(id -u):$(id -g)"
		"ethereum/solc:${solidity_version}"
	)
	"${solc_exec[@]}" --abi "${solidity_file_name}" -o contracts/abi/ --overwrite
	"${solc_exec[@]}" --bin "${solidity_file_name}" -o contracts/bin/ --overwrite
```